internal/resolution/entitysources/catalogdsource.go

```diff
@@ -0,0 +1,157 @@
+package entitysources
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/operator-framework/deppy/pkg/deppy"
+	"github.com/operator-framework/deppy/pkg/deppy/input"
+	"github.com/operator-framework/operator-registry/alpha/property"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	catalogd "github.com/operator-framework/catalogd/pkg/apis/core/v1beta1"
+)
+
+// catalogdEntitySource is a source for(/collection of) deppy defined input.Entity, built from content
+// made accessible on-cluster by https://github.com/operator-framework/catalogd.
+// It is an implementation of deppy defined input.EntitySource
+type catalogdEntitySource struct {
+	client client.Client
+}
+
+func NewCatalogdEntitySource(ctx context.Context, client client.Client) *catalogdEntitySource {
```

Comment:
Can't tell if my comment was lost. But this `ctx` parameter is no longer necessary in this function signature.